### PR TITLE
RUMM-441 Update the APM and RUM features docs

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogConfig.kt [hunk 1]
```diff
@@ -290,9 +290,11 @@ private constructor(
 
         /**
          * Sets the automatic view tracking strategy used by the SDK.
-         * By default this is null.
+         * By default this feature is disabled.
```
Comment (xgouchet) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogConfig.kt:293]: By default, no view will be tracked.

File: dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogConfig.kt [hunk 2]
```diff
@@ -290,9 +290,11 @@ private constructor(
...
          * @param strategy as the [ViewTrackingStrategy]
-         * (e.g. FragmentViewTrackingStrategy, ActivityViewTrackingStrategy)
+         * @see [com.datadog.android.rum.tracking.ActivityViewTrackingStrategy]
+         * @see [com.datadog.android.rum.tracking.FragmentViewTrackingStrategy]
```
Comment (xgouchet) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogConfig.kt:296]: + NavigationViewTrackingStrategy

File: docs/rum_collection.md [hunk 3]
```diff
@@ -63,20 +63,24 @@ class SampleApplication : Application() {
...
 Depending on your application's architecture, you can choose one of several implementations of `ViewTrackingStrategy`:
- 
+
   - `ActivityViewTrackingStrategy`: Every activity in your application is considered a distinct view.
   - `FragmentViewTrackingStrategy`: Every fragment in your application is considered a distinct view.
   - `NavigationViewTrackingStrategy`: If you use the Android Jetpack Navigation library, this is the recommended strategy. It automatically tracks the navigation destination as a distinct view.
+  - `MixedViewTrackingStrategy`: Every activity or fragment in your application is considered a distinct view. This strategy is a mix between the `ActivityViewTrackingStrategy` and `FragmentViewTrackingStrategy`.
+
+
+  **Note**: For the `ActivityViewTrackingStrategy`, `FragmentViewTrackingStrategy` or `MixedViewTrackingStrategy` that are more general strategies you will be able to validate which `Fragment` or `Activity` to be tracked as a RUM View event by providing a `ComponentPredicate` implementation in the constructor.
```
Comment (xgouchet) [inline docs/rum_collection.md:75]: you don't need the "that are more general strategies" part.

File: docs/trace_collection.md [hunk 4]
```diff
@@ -84,14 +84,104 @@ class SampleApplication : Application() {
...
+             }
+             catch(e:Error){
+               childSpan.error(e)
+             }
+          }
+        }
+        catch(e:Error){
+          span.error(e)
+        }
+   }
```
Comment (xgouchet) [inline docs/trace_collection.md:111]: You still need to close the span manually, in a finally clause maybe

File: docs/trace_collection.md [hunk 5]
```diff
@@ -84,14 +84,104 @@ class SampleApplication : Application() {
...
+   ```  
+9. How to distribute traces between your environments (e.g. frontend - backend):
```
Comment (xgouchet) [inline docs/trace_collection.md:142]: They don't need to do that by hand, there's the Tracing Interceptor for that

File: docs/rum_collection.md [hunk 6]
```diff
@@ -63,20 +63,24 @@ class SampleApplication : Application() {
...
 Depending on your application's architecture, you can choose one of several implementations of `ViewTrackingStrategy`:
- 
+
   - `ActivityViewTrackingStrategy`: Every activity in your application is considered a distinct view.
   - `FragmentViewTrackingStrategy`: Every fragment in your application is considered a distinct view.
   - `NavigationViewTrackingStrategy`: If you use the Android Jetpack Navigation library, this is the recommended strategy. It automatically tracks the navigation destination as a distinct view.
+  - `MixedViewTrackingStrategy`: Every activity or fragment in your application is considered a distinct view. This strategy is a mix between the `ActivityViewTrackingStrategy` and `FragmentViewTrackingStrategy`.
+
+
+  **Note**: For the `ActivityViewTrackingStrategy`, `FragmentViewTrackingStrategy` or `MixedViewTrackingStrategy` you will be able to validate which `Fragment` or `Activity` to be tracked as a RUM View event by providing a `ComponentPredicate` implementation in the constructor.
```
Comment (ruthnaebeck) [inline docs/rum_collection.md:75]:
```suggestion
  **Note**: For `ActivityViewTrackingStrategy`, `FragmentViewTrackingStrategy`, or `MixedViewTrackingStrategy` you can validate which `Fragment` or `Activity` is tracked as a RUM view event by providing a `ComponentPredicate` implementation in the constructor.
```

File: docs/trace_collection.md [hunk 7]
```diff
@@ -84,14 +84,108 @@ class SampleApplication : Application() {
...
+   }
+   catch(e:Error){
+     span.error(e)
+   }
+   finally{
+     span.finish()
+   }
+
+   ```  
+9. (Optional) How to manually distribute traces between your environments (e.g. frontend - backend):
```
Comment (ruthnaebeck) [inline docs/trace_collection.md:144]:
```suggestion
9. (Optional) How to manually distribute traces between your environments, for example frontend - backend:
```

File: docs/trace_collection.md [hunk 8]
```diff
@@ -84,14 +84,108 @@ class SampleApplication : Application() {
...
+
+   * Step 1: Inject Tracer context in the client request
```
Comment (ruthnaebeck) [inline docs/trace_collection.md:146]:
```suggestion
   * Step 1: Inject tracer context in the client request.
```

File: docs/trace_collection.md [hunk 9]
```diff
@@ -84,14 +84,108 @@ class SampleApplication : Application() {
...
+   tracer.inject(
+           span.context(),
+           Format.Builtin.TEXT_MAP_INJECT,
+           TextMapInject { key, value ->
+               tracedRequestBuilder.addHeader(key, value)
+           }
+    val HttpRequest request = tracedRequestBuilder.build()
+    // Dispatch the request and close finish the span after
+   ```
+   * Step 2: Extract the client Tracer context from headers in server code
```
Comment (ruthnaebeck) [inline docs/trace_collection.md:161]:
```suggestion
   * Step 2: Extract the client tracer context from headers in server code.
```

File: docs/trace_collection.md [hunk 10]
```diff
@@ -84,14 +84,108 @@ class SampleApplication : Application() {
...
+   val span = tracer.buildSpan("<SPAN_NAME>").start()
+   val tracedRequestBuilder = Request.Builder() ...
+   tracer.inject(
+           span.context(),
+           Format.Builtin.TEXT_MAP_INJECT,
+           TextMapInject { key, value ->
+               tracedRequestBuilder.addHeader(key, value)
+           }
+    val HttpRequest request = tracedRequestBuilder.build()
+    // Dispatch the request and close finish the span after
```
Comment (ruthnaebeck) [inline docs/trace_collection.md:159]:
```suggestion
    // Dispatch the request and finish the span after.
```

File: docs/trace_collection.md [hunk 11]
```diff
@@ -84,14 +84,108 @@ class SampleApplication : Application() {
...
+                    .toMap()
+                    .toMutableMap()
+                    .iterator()
+            }
+        )
+   val serverSpan = tracer.buildSpan("<SERVER_SPAN_NAME>").asChildOf(extractedContext).start()      
+
+   ```
+
+**Note**: For code bases using the OkHttp client we provide a more straight forward implementation below.
```
Comment (ruthnaebeck) [inline docs/trace_collection.md:181]:
```suggestion
**Note**: For code bases using the OkHttp client, Datadog provides the implementation below.
```

File: docs/trace_collection.md [hunk 12]
```diff
@@ -84,14 +84,108 @@ class SampleApplication : Application() {
...
+
+10. (Optional) - Provide additional tags alongside your span.
 
     ```kotlin
     span.setTag("http.url", url)
     ```
-7. (Optional) Attach an error information to a Span:
+11. (Optional) Attach an error information to a Span:
```
Comment (ruthnaebeck) [inline docs/trace_collection.md:188]:
```suggestion
11. (Optional) Attach an error information to a span:
```

**Pull request merged**